database/merkledb/unit.go

```diff
@@ -0,0 +1,140 @@
+package merkledb
+
+const UnitSize = 16 + 1
+
+type Unit byte
+
+// FirstNonPrefix returns the first Unit that's not common between baseKey and otherKey
+func FirstNonPrefix(baseKey []Unit, otherKey []Unit) Unit {
+	smaller := otherKey
+	larger := baseKey
+
+	// the UnitSize - 1 position is the termination position
+	// that position will only have LeafNodes
```

Comment:
I don't understand what this means...can you pls explain?

Reply:
If the len of the basekey is the same as the otherkey it will have no prefix.

It will be the same address. And in a BranchNode, the same address means that it's a LeafNode with the Address of the BranchNode. Will set those Nodes at the end of the BranchNode array in position `UnitSize - 1` .

Reply:
I think this is answered in the comment above :)